### PR TITLE
Feat | add error selectors

### PR DESCRIPTION
**Pull request by adisreyaj**
## Description
Add `role=alert` and `data-alert-type="failure"` for components showing error states:
1. Toast Error
2. Load Async Error 
3. Not Found

Selector:
```js
$('[role=alert][data-alert-type="failure"]')
```

### CODE REVIEW

File: projects/components/src/load-async/wrapper/load-async-wrapper.component.ts [hunk 1]
```diff
@@ -21,6 +21,9 @@ export const ASYNC_WRAPPER_PARAMETERS$ = new InjectionToken<Observable<LoadAsync
       </ng-container>
       <ng-container *ngSwitchDefault>
         <ht-message-display
+          aria-live="polite"
+          role="alert"
+          [attr.data-alert-type]="state.type === '${LoadAsyncStateType.GenericError}' ? 'failure' : 'no-data'"
```
Comment (arjunlalb) [inline projects/components/src/load-async/wrapper/load-async-wrapper.component.ts:26]: can we use the `NotificationMode` enum here for the failure string?

File: projects/components/src/not-found/not-found.component.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ import { ButtonRole, ButtonStyle } from '../button/button';
   styleUrls: ['./not-found.component.scss'],
   template: `
     <div class="not-found-container fill-container">
-      <div class="not-found-content">
+      <div class="not-found-content" role="alert" aria-live="assertive" [attr.data-alert-type]="'failure'">
```
Comment (arjunlalb) [inline projects/components/src/not-found/not-found.component.ts:12]: same here.

File: projects/components/src/load-async/wrapper/load-async-wrapper.component.ts (hunk 1 above)
Comment (adisreyaj) [reply projects/components/src/load-async/wrapper/load-async-wrapper.component.ts]: Updated.

File: projects/components/src/not-found/not-found.component.ts (hunk 2 above)
Comment (adisreyaj) [reply projects/components/src/not-found/not-found.component.ts]: Updated.

**Review by anandtiwary — APPROVED** [review]

**Pull request merged**